src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,146 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(head, value, tail);
```

Comment:
you're adding value to the end of the list. so new node shouldn't have tail since it will become tail. and prev node should be tail. so, 17, 19 lines will not be needed